File: scripts/generate-indexes.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import os
import json
from pathlib import Path

RECETARIO_DIR = Path(__file__).resolve().parent.parent / 'data' / 'recetario'

def is_image(name):
    return name.lower().endswith(('.jpg','.jpeg','.png','.gif','.bmp','.webp'))

def is_pdf(name):
    return name.lower().endswith('.pdf')

def generate_indexes():
    if not RECETARIO_DIR.exists():
        print('Error: no existe', RECETARIO_DIR)
        return

    for entry in sorted(RECETARIO_DIR.iterdir()):
        if not entry.is_dir():
            continue
        folder = entry
        try:
            files = sorted([f.name for f in folder.iterdir() if f.is_file()])
        except Exception as e:
            print('No se pudo leer', folder, e)
            continue

        images = [f for f in files if is_image(f)]
        pdfs = [f for f in files if is_pdf(f)]

        index = {
            'folder': folder.name,
            'files': files,
            'cover': images[0] if images else None,
            'pdf': pdfs[0] if pdfs else None,
            'generatedAt': __import__('datetime').datetime.utcnow().isoformat() + 'Z'
        }

        out = folder / 'index.json'
        try:
            with open(out, 'w', encoding='utf-8') as fh:
                json.dump(index, fh, ensure_ascii=False, indent=2)
            print('Wrote', out, 'cover=', index['cover'], 'pdf=', index['pdf'])
        except Exception as e:
            print('Error escribiendo', out, e)

if __name__ == '__main__':
    generate_indexes()

File: scripts/generate-indexes.js
const fs = require('fs');
const path = require('path');

const recetarioPath = path.join(__dirname, '..', 'data', 'recetario');

function isImage(name) {
  return /\.(jpe?g|png|gif|bmp|webp)$/i.test(name);
}

function isPDF(name) {
  return /\.pdf$/i.test(name);
}

function generate() {
  const folders = fs.readdirSync(recetarioPath, { withFileTypes: true })
    .filter(d => d.isDirectory())
    .map(d => d.name);

  folders.forEach(folder => {
    const folderPath = path.join(recetarioPath, folder);
    let files = [];

    try {
      files = fs.readdirSync(folderPath);
    } catch (err) {
      console.warn('No se puede leer', folderPath, err.message);
      return;
    }

    // Orden estable: imágenes primero, pdfs after, then others
    const images = files.filter(isImage);
    const pdfs = files.filter(isPDF);
    const others = files.filter(f => !isImage(f) && !isPDF(f));

    const index = {
      folder: folder,
      files: files.sort(),
      cover: images.length ? images[0] : null,
      pdf: pdfs.length ? pdfs[0] : null,
      generatedAt: new Date().toISOString()
    };

    const outPath = path.join(folderPath, 'index.json');
    fs.writeFileSync(outPath, JSON.stringify(index, null, 2), 'utf8');
    console.log('Wrote', outPath, 'cover=', index.cover, 'pdf=', index.pdf);
  });
}

generate();
